agent/src/main/java/com/teamscale/jacoco/agent/commit_resolution/git_properties/GitPropertiesLocatorUtils.java

```diff
@@ -97,19 +129,47 @@ public static ProjectRevision getProjectRevisionFromGitProperties(
 	}
 
 	/** Returns a pair of the zipfile entry name and parsed properties, or null if no git.properties were found. */
-	public static Pair<String, Properties> findGitPropertiesInJar(
-			File jarFile) throws IOException {
+	public static Pair<String, Properties> findGitPropertiesInFile(
+			File file, boolean isJarFile) throws IOException {
+		if (isJarFile) {
+			return findGitPropertiesInJarFile(file);
+		}
+		return findGitPropertiesInDirectoryFile(file);
+	}
+
+	private static Pair<String, Properties> findGitPropertiesInJarFile(File file) throws IOException {
 		try (JarInputStream jarStream = new JarInputStream(
-				new BashFileSkippingInputStream(new FileInputStream(jarFile)))) {
-			return findGitPropertiesInJar(jarStream);
+				new BashFileSkippingInputStream(new FileInputStream(file)))) {
+			return findGitPropertiesInJarFile(jarStream);
 		} catch (IOException e) {
-			throw new IOException("Reading jar " + jarFile.getAbsolutePath() + " for obtaining commit " +
+			throw new IOException("Reading jar " + file.getAbsolutePath() + " for obtaining commit " +
 					"descriptor from git.properties failed", e);
 		}
 	}
 
+	private static Pair<String, Properties> findGitPropertiesInDirectoryFile(File file) throws IOException {
+		File[] directoryFiles = file.listFiles();
```

Comment:
That will only list direct children, right? I guess we need a recursive solution here. 

`FileSystemUtils.listFilesRecursively` might be of help here

Reply:
Yes, I have actually assumed the git.properties file should always be at the top. If this is not the case, then we need the recursive solution.